projects/sunbird-epub-player/src/lib/sunbird-epub-player.component.ts

```diff
@@ -1,15 +1,23 @@
-import { Component, OnInit } from '@angular/core';
+import {  EventEmitter ,Component, Output, Input, OnInit } from '@angular/core';
+import { PlayerConfig } from './sunbird-epub-player.interface';
 
 @Component({
   selector: 'sunbird-epub-player',
   templateUrl: './sunbird-epub-player.component.html',
   styles: []
 })
 export class EpubPlayerComponent implements OnInit {
+  @Input() playerConfig: PlayerConfig;
+  epubSrc: String;
+  @Output() headerActionsEvent : EventEmitter<any> = new EventEmitter<any>();
 
   constructor() { }
 
   ngOnInit() {
+    this.epubSrc = this.playerConfig.metadata.streamingUrl;
```

Comment:
Check for streaming support and  offline error events
fallback url to artifact if streamingUrl is not present